### DISCUSSION TITLE
ENHANCED_RESOURCE_SCOPING and Istio upgrade issue

### DISCUSSION THREAD

**Discussion by JacobAmar** [istio/istio#51757] [Q&A]
Hello everyone,

I recently went through an issue after upgrading my Istio version to 1.22.1 from 1.21.1 and would appreciate your insights.

Here's a quick overview of my setup: I have two control planes, both located within the istio-system namespace:

1. The "default" control plane, which is able to discover all namespaces within the cluster.
2. The "test-namespace-control-plane" intended for reducing load, it only discovers namespaces with the "test-namespace-control-plane: true" label.

Everything was functioning well with version 1.21.1. However, after the upgrade, I began encountering a 400 error whenever I tried to send requests to my service through the Istio Ingress Gateway. The Istio access logs flagged a dpe http1.codec_error.

Upon reviewing the changelog for the 1.22.1 update, I noticed that the ENHANCED_RESOURCE_SCOPING was set to 'true'. Interestingly, when I switched this variable to 'false', the service started functioning correctly again.

I am seeking clarification on two aspects. 
1. I'd like to understand the purpose and role of the ENHANCED_RESOURCE_SCOPING environment variable.
2. I would appreciate feedback on whether my setup is correctly configured with this variable set to 'false'. It is not clear to me why this change was necessary for the system to work again.

I'm looking forward to any insights or suggestions you can provide.

Thank you for your help.

**Comment by howardjohn** [comment]
ENHANCED_RESOURCE_SCOPING makes the selection apply to all resources, rather than just  few of them. So presumably you depend on some resources you are filtering out
